lib/elixir/lib/access.ex

```diff
@@ -610,6 +625,8 @@ defmodule Access do
       ** (RuntimeError) Access.at/1 expected a list, got: %{}
 
   """
+  @spec at(non_neg_integer) :: (:get | :get_and_update, list, fun -> {list, list})
```

Comment:
This type is leaking implementation detail. I don't think most people look at this spec and say "I understand what it is doing". We should probably abstract away the `get | get_and_update` function into a type.

Also note that the first element in `{list, list}` is not required to be a list.